johnwayodi/sm-server
# #161695595 Restrucuture API for easy exchange with Front-End

#### What does this PR do?
Change some of the Endpoint return values to enable painless integration with Front-End.
Add Flask-cors to allow accessibility of API endpoints.
#### Description of Task to be completed?
1. Change GET all users endpoint to return nested JSON array object instead of nested dictionaries.
2. Change GET single sale endpoint to return nested JSON array object instead of nested dictionaries.
3. Change GET all products to return nested JSON array object instead of nested dictionaries.
4. Fix Sales anomaly.
5. Remove unused imports and unused configurations.
6. Move configurations from `run.py` to appropriate location
7. Refactor sample test data to incorporate changed API endpoints responses 
#### How should this be manually tested?
1. `git clone https://github.com/johnwayodi/sm-api-v2.git`
2. Ensure you are in the cloned repo, **sm-api-v2** by default.
3. Create a virtual environment using virtualenv and `pip install -r requirements.txt`
4. `git checkout -b ch-restructure-api-161695595 origin/ch-restructure-api-161695595`
5. Run **_pytest -vv --cov=storemanager/api/v2_** in terminal to run the tests and get coverage on the application.
#### Any background context you want to provide?
No background context
#### What are the relevant pivotal tracker stories?
[#161695595](https://www.pivotaltracker.com/story/show/161695595)
#### Questions:
None for this section
